pkg/platform/kube/functionres/lazy.go

```diff
@@ -1163,9 +1158,23 @@ func (lc *lazyClient) createOrUpdateCronJob(functionLabels labels.Set,
 	cronJobSpec.Suspend = &suspendCronJob
 
 	getCronJob := func() (interface{}, error) {
-		return lc.kubeClientSet.BatchV1beta1().
+		cronJobs, err := lc.kubeClientSet.BatchV1beta1().
 			CronJobs(function.Namespace).
-			Get(kube.CronJobNameFromFunctionName(function.Name, jobName), metav1.GetOptions{})
+			List(metav1.ListOptions{
+				LabelSelector: fmt.Sprintf("nuclio.io/function-cron-trigger-cron-job=true,"+
```

Comment:
dry: move the label selector to a compile function (e.g.: `compileCronJobLabelSelector`) and reuse